source/slang/slang-type-layout.cpp

```diff
@@ -3183,6 +3183,35 @@ RefPtr<VarLayout> StructTypeLayoutBuilder::addField(
     return fieldLayout;
 }
 
+RefPtr<VarLayout> StructTypeLayoutBuilder::addExplicitUniformField(DeclRef<VarDeclBase> field, TypeLayoutResult fieldResult)
+{
+    auto packoffsetModifier = field.getDecl()->findModifier<HLSLPackOffsetSemantic>();
+    if (!packoffsetModifier)
+        return nullptr;
+
+    RefPtr<VarLayout> fieldLayout = new VarLayout();
+    fieldLayout->varDecl = field;
+    fieldLayout->typeLayout = fieldResult.layout;
+    m_typeLayout->fields.add(fieldLayout);
+    if (field)
+    {
+        m_typeLayout->mapVarToLayout.add(field.getDecl(), fieldLayout);
+    }
+    UInt uniformOffset = packoffsetModifier->uniformOffset;
+    if (fieldResult.layout->FindResourceInfo(LayoutResourceKind::Uniform))
+    {
+        fieldLayout->AddResourceInfo(LayoutResourceKind::Uniform)->index = uniformOffset;
+    }
+    UniformLayoutInfo fieldInfo = fieldResult.info.getUniformLayout();
+    auto uniformInfo = m_info;
+    m_rules->AddStructField(&uniformInfo, fieldInfo);
+    m_info.alignment = uniformInfo.alignment;
+    m_info.size.raw = Math::Max(
+        m_info.size.getFiniteValue(),
+        (size_t)(uniformOffset + fieldResult.layout->FindResourceInfo(LayoutResourceKind::Uniform)->count.getFiniteValue()));
```

Comment:
Should this be rounded to a multiple of `alignment`?

Reply:
I think we/dxc don't round sizes up to alignment.